loopy/transform/compute.py

```diff
@@ -0,0 +1,196 @@
+import islpy as isl
+
+import loopy as lp
+from loopy.kernel import LoopKernel
+from loopy.kernel.data import AddressSpace
+from loopy.kernel.function_interface import CallableKernel, ScalarCallable
+from loopy.match import parse_stack_match
+from loopy.symbolic import (
+    RuleAwareSubstitutionMapper,
+    SubstitutionRuleMappingContext,
+    pw_aff_to_expr
+)
+from loopy.translation_unit import TranslationUnit
+
+from pymbolic import var
+from pymbolic.mapper.substitutor import make_subst_func
+
+from pytools.tag import Tag
+
+
+def compute(
+        t_unit: TranslationUnit,
+substitution: str,
+        *args,
+        **kwargs
+    ) -> TranslationUnit:
+    """
+    Entrypoint for performing a compute transformation on all kernels in a
+    translation unit. See :func:`_compute_inner` for more details.
+    """
+
+    assert isinstance(t_unit, TranslationUnit)
+    new_callables = {}
+
+    for id, callable in t_unit.callables_table.items():
+        if isinstance(callable, CallableKernel):
+            kernel = _compute_inner(
+                callable.subkernel,
+                substitution,
+                *args, **kwargs
+            )
+
+            callable = callable.copy(subkernel=kernel)
+        elif isinstance(callable, ScalarCallable):
+            pass
+        else:
+            raise NotImplementedError()
+
+        new_callables[id] = callable
+
+    return t_unit
+
+def _compute_inner(
```

Comment:
IMO, this would've been an opportunity to try out namedisl, at least locally, to see how the interface "feels".